components/mailgun/mailgun.app.js

```diff
@@ -0,0 +1,192 @@
+      default: "US",
+      options: ["US", "EU"],
+    },
+    timer: {
+      label: "Polling schedule",
+      description: "Pipedream polls Reddit for events on this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15, // by default, run every 15 minutes.
+      },
+    },
+  },
+  methods: {
+    mailgun(apiKey, domain, host) {
+      return require("mailgun-js")({ apiKey, domain, host });
+    },
+    _apiKey() {
+      return this.$auth.api_key;
+    },
+    _apiHost() {
+      const subdomain = this.baseRegion == "EU" ? "api.eu" : "api";
+      return `${subdomain}.mailgun.net`;
+    },
+    async _makeRequest(opts) {
+      const mailgun = this.mailgun(
+        this._apiKey(),
+        opts.domain,
+        this._apiHost()
+      );
+      switch (opts.method) {
+        case "get":
+          return mailgun.get(opts.path, opts.params);
+        case "post":
+          return mailgun.post(opts.path, opts.params);
+        case "put":
+          return mailgun.put(opts.path, opts.params);
+        case "delete":
+          return mailgun.delete(opts.path, opts.params);
+        default:
+          return;
+      }
+    },
+    _isRetriableStatusCode(statusCode) {
+      [402, 429, 500, 502, 503, 504].includes(statusCode);
+      //More info on  Mailgun HTTP response codes: https://documentation.mailgun.com/en/latest/api-intro.html?highlight=502#status-codes
+    },
+    async _withRetries(apiCall, allow404 = false) {
+      const retryOpts = {
+        retries: 5,
+        factor: 2,
+      };
+      return retry(async (bail) => {
+        try {
+          return await apiCall();
+        } catch (err) {
+          const statusCode = get(err, ["response", "status"]);
+          if (!this._isRetriableStatusCode(statusCode)) {
+            bail(`
+              Unexpected error (status code: ${statusCode}):
+              ${JSON.stringify(err.response)}
+            `);
+          }
+          console.warn(`Temporary error: ${err.message}`);
+          throw err;
+        }
+      }, retryOpts);
+    },
+    async getDomains(skip) {
+      return await this._withRetries(() =>
+        this._makeRequest({
+          method: "get",
+          path: `/domains`,
+          params: {
+            skip,
+          },
+        })
+      );
+    },
+    async getWebhookDetails(mailgunDomain, webhookName) {
+      const webhooks = await this._withRetries(() =>
+        this._makeRequest({
+          method: "get",
+          path: `/domains/${mailgunDomain}/webhooks`,
+        })
+      );
+      return get(webhooks, `webhooks.${webhookName}`);
+    },
+    async getAllDomains() {
+      let domains_set;
+      const results = [];
+      let skip = 0;
+      do {
+        domains_set = await this.getDomains(skip);
+        domains_set.items.forEach((domain) => {
```

Comment:
You can shorten this code to something like:

```javascript
results.push(...domains_set.items.map(domain => domain.name))
```

Reply:
All right, I made the change.

Reply:
Sorry I need to take that back. I made a concept test with two array on this [workflow ](https://pipedream.com/@sergio/javascript-playground-p_5VCL7by/edit)and it works there. I don't know why it didn't work here. I tried a couple of variations such as declaring an anonymous function and pass it as argument to the map function.